src/routing/IRouter.ts

```diff
@@ -1,15 +1,27 @@
 import { Controller } from '../controller/controller'
 import { Middleware } from '../middleware/middleware'
-import { Request } from '../request'
+import { RequestEvent } from '../request/request-event'
 import { Route } from './routes'
 
-export interface RouteData {
-  route: Route<Middleware, Controller>
-  params: string[]
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+
+export type RouteMetadata<
+  TRoute extends Route<Middleware, Controller> = Route<Middleware, Controller>
+> = Omit<TRoute, 'controller' | 'middleware'>
+
+export interface RouteData<
+  TRoute extends Route<Middleware, Controller> = Route<Middleware, Controller>
+> {
+  route: TRoute
+  metadata: RouteMetadata<TRoute>
+  pathParameters: { [key: string]: string }
```

Comment:
Time 4 map?

Reply:
could do that.. will need to somewhere translate them to maps, is it worth it?  I suppose?